staging/src/k8s.io/apiserver/pkg/server/hooks.go

```diff
@@ -113,26 +147,54 @@ func (s *GenericAPIServer) RunPostStartHooks(stopCh <-chan struct{}) {
 	}
 }
 
-// isHookRegistered checks whether a given hook is registered
-func (s *GenericAPIServer) isHookRegistered(name string) bool {
+// RunPreStopHooks runs the PreStopHooks for the server
+func (s *GenericAPIServer) RunPreStopHooks() {
+	glog.Infof("Running PreStopHooks: %v", s.preStopHooks)
+	s.preStopHookLock.Lock()
+	defer s.preStopHookLock.Unlock()
+	s.preStopHooksCalled = true
+
+	context := HookContext{
+		LoopbackClientConfig: s.LoopbackClientConfig,
+	}
+
+	for hookName, hookEntry := range s.preStopHooks {
+		runPreStopHook(hookName, hookEntry, context)
+	}
+}
```

Comment:
this must block. Does it?

Reply:
Yes.  Unlike the post-start hooks, these aren't run in a goroutine